task/src/lib.rs

```diff
@@ -407,19 +407,8 @@ pub mod pallet {
 
 		// Function to transfer balance from one account to another
 		#[transactional]
-		pub fn transfer_balance(task_initiator: &T::AccountId, task_volunteer: &T::AccountId, budget: BalanceOf<T>) -> Result<(), Error<T>> {
-			// TODO: Investigate why Currency transfer doesn't work 
-			// TODO: See proper testing https://docs.substrate.io/how-to-guides/v3/testing/transfer-function/
-
-			//let task_initiator = task.initiator.clone();
-			log::info!("budget {:?}.", budget);
-			log::info!("signer {:?}.", task_initiator);
-			log::info!("task_volunteer {:?}.", task_volunteer);
-			ensure!(<T as self::Config>::Currency::free_balance(task_initiator) >= budget, <Error<T>>::NotEnoughBalance);
-			let result = <T as self::Config>::Currency::transfer(task_initiator, task_volunteer, budget, ExistenceRequirement::KeepAlive);
-			log::info!("Result Output {:?}.", result);
-
-			Ok(())
+		pub fn transfer_balance(task_initiator: &T::AccountId, task_volunteer: &T::AccountId, budget: BalanceOf<T>) -> Result<(),  DispatchError> {
```

Comment:
Claned up all the logs and comments.. Nice!

Reply:
@JosephKnecht-lab `transfer_balance` does not do anything special then what `Currency::transfer` is doing. Is there any reason/purpose for this function which I am not able to see?